### PR TITLE
bump kube deps to 1.27.4

### PR DESCRIPTION
**Pull request by sanchezl**
Get CI going again and bump kube deps to 1.27.4

### CODE REVIEW

**Review by dinhxuanvu — COMMENTED** [review]
/lgtm

File: go.mod [hunk 1]
```diff
@@ -1,74 +1,76 @@
...
 	github.com/spf13/pflag v1.0.5
-	k8s.io/api v0.21.0
-	k8s.io/apiextensions-apiserver v0.21.0
-	k8s.io/apimachinery v0.21.0
-	k8s.io/client-go v0.21.0
-	k8s.io/code-generator v0.21.0
-	k8s.io/klog/v2 v2.20.0
-	k8s.io/kube-aggregator v0.21.0
+	k8s.io/api v0.27.4
+	k8s.io/apiextensions-apiserver v0.24.4
```
Comment (benluddy) [inline go.mod:12]: Missed this one?

File: cmd/trigger/app/trigger.go [hunk 2]
```diff
@@ -66,6 +66,7 @@ func Run(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
+	migration.DiscoveryClient.UseLegacyDiscovery = true
```
Comment (benluddy) [inline cmd/trigger/app/trigger.go:69]: Why is this necessary? Shouldn't content-type negotiation handle compatibility with apiservers that don't support aggregation?

File: cmd/initializer/Dockerfile [hunk 3]
```diff
@@ -11,8 +11,12 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+FROM golang:1.19.2 as builder
```
Comment (benluddy) [inline cmd/initializer/Dockerfile:14]: `1.20`?

And is there any concern about the docker.io rate limit affecting CI jobs? I see a few other places within the kubernetes org that depend on this repo, but I don't have context on any of those.

**Review by benluddy — CHANGES_REQUESTED** [review]

File: cmd/trigger/app/trigger.go (hunk 2 above)
Comment (sanchezl) [reply cmd/trigger/app/trigger.go:69]: I needed this here to ensure I [get the `StorageVersionHash` field populated](https://github.com/kubernetes/enhancements/blob/98e89c38b263e43755d3b8cd20e4083f9f71af02/keps/sig-api-machinery/3352-aggregated-discovery/README.md?plain=1#L782-L787).

File: cmd/initializer/Dockerfile (hunk 3 above)
Comment (sanchezl) [reply cmd/initializer/Dockerfile]: Thanks Ben. 
As for docker.io rate limits, this particular repo doesn't get built too often. I tried searching for an alternative on registry.k8s.io, but nothing stood out.

File: go.mod (hunk 1 above)
Comment (sanchezl) [reply go.mod]: Done.

**Comment by deads2k** [comment]
> migration.DiscoveryClient.UseLegacyDiscovery = true

Create issue to fix this.
